src/app/services/cache.service.ts

```diff
@@ -0,0 +1,66 @@
+import Redis from 'ioredis';
+
+let client: Redis | null = null;
+
+export function getClient(): Redis {
```

Comment:
The cache might use more encapsulation (mentioned in previous comment) - it should not directly expose Redis and it should encapsulate the client itself so that the callers dont have to pass in redis client with each method call.